src/event.c

```diff
@@ -314,23 +381,53 @@ bool event_loop(void) {
 	running = true;
 
 #ifndef HAVE_MINGW
+
+#ifdef HAVE_SYS_EPOLL_H
+
+	if(!epollset) {
+		epollset = event_epoll_init();
+	}
+
+#else
 	fd_set readable;
 	fd_set writable;
+#endif
 
 	while(running) {
 		struct timeval diff;
-		struct timeval *tv = get_time_remaining(&diff);
+		struct timeval *tv = timeout_execute(&diff);
+#ifndef HAVE_SYS_EPOLL_H
 		memcpy(&readable, &readfds, sizeof(readable));
 		memcpy(&writable, &writefds, sizeof(writable));
+#endif
+
+
+#ifdef HAVE_SYS_EPOLL_H
+		struct epoll_event events[EPOLL_MAX_EVENTS_PER_LOOP];
+		long timeout = (tv->tv_sec * 1000) + (tv->tv_usec / 1000);
 
-		int fds = 0;
+		if(timeout > INT_MAX) {
+			timeout = INT_MAX;
+		}
+
+		int n = epoll_wait(epollset, events, EPOLL_MAX_EVENTS_PER_LOOP, (int)timeout);
+#else
+		int maxfds = 1;
```

Comment:
Zero is fine if there really are no fds set.

Reply:
Done